app/javascript/mastodon/features/interaction_modal/index.jsx

```diff
@@ -127,18 +127,36 @@ class InteractionModal extends PureComponent {
     }
 
     let signupButton;
+    let signUpOrSignInButton;
 
-    if (registrationsOpen) {
-      signupButton = (
-        <a href={signupUrl} className='button button--block button-tertiary'>
-          <FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' />
+    if (sso_redirect) {
+      signUpOrSignInButton = (
+        <a href={sso_redirect} data-method='post' className='button button--block button-tertiary'>
+          <FormattedMessage id='sign_in_banner.sso_redirect' defaultMessage='Login or Register' />
         </a>
-      );
+      )
     } else {
-      signupButton = (
-        <button className='button button--block button-tertiary' onClick={this.handleSignupClick}>
-          <FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' />
-        </button>
+      if(registrationsOpen) {
+        signupButton = (
+          <a href={signupUrl} className='button button--block button-tertiary'>
+            <FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' />
+          </a>
+        );
+      } else {
+        signupButton = (
+          <button className='button button--block button-tertiary' onClick={this.handleSignupClick}>
+            <FormattedMessage id='sign_in_banner.create_account' defaultMessage='Create account' />
+          </button>
+        );
+      }
+
+      signUpOrSignInButton = (
+        <>
+        <a href='/auth/sign_in' className='button button--block'>
+          <FormattedMessage id='sign_in_banner.sign_in' defaultMessage='Login' />
+        </a>
+        {signupButton}
+        </>
```

Comment:
Wrong indentation here.

```suggestion
        <>
          <a href='/auth/sign_in' className='button button--block'>
            <FormattedMessage id='sign_in_banner.sign_in' defaultMessage='Login' />
          </a>
          {signupButton}
        </>
```